packages/core/getUserAvailability.ts

```diff
@@ -139,6 +145,58 @@ export async function getUserAvailability(
     source: query.withSource ? a.source : undefined,
   }));
 
+  if (bookingLimits) {
+    // Get all dates between dateFrom and dateTo
+    const dates = []; // this is as dayjs date
+    let startDate = dayjs(dateFrom);
+    const endDate = dayjs(dateTo);
+    while (startDate.isBefore(endDate)) {
+      dates.push(startDate.add(1, "day"));
+      startDate = startDate.add(1, "day");
+    }
+
+    const ourBookings = busyTimes.filter((busyTime) =>
+      busyTime.source?.startsWith(`eventType-${eventType?.id}`)
+    );
+
+    // Apply booking limit filter against our bookings
+
+    for (const [key, limit] of Object.entries(bookingLimits)) {
+      const limitKey = key as keyof BookingLimit;
+
+      if (limitKey === "PER_YEAR") {
```

Comment:
If PER_YEAR we use DB call to check - as availability only takes in the busy events for that month